### PR TITLE
JBPM-8118: Containment is never allowed when multiple selection.

### PR DESCRIPTION
**Pull request by romartin**
Hey @hasys @tiagodolphine 

This issue fixes [JBPM-8118](https://issues.jboss.org/browse/JBPM-8118) and some other issue I found during testing.

This commit does:
- Created `WiresLayerIndex` type in order to generify the index for wires shapes in the layer
- Use of shared indexes on composite handler/control, instead of single a instance for each one. It fixes some issues and also improves performance
- Fixed issues related to ControlPoints (commands, control, acceptor)

Assembly:
https://github.com/kiegroup/kie-wb-common/pull/2451
https://github.com/kiegroup/lienzo-tests/pull/63

Thanks!

### CODE REVIEW

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresShapeControlImpl.java [hunk 1]
```diff
@@ -460,7 +482,10 @@ private WiresManager getWiresManager() {
...
     }
 
-    private ColorMapBackedPicker getPicker() {
-        return parentPickerControl.getPicker();
+    private void moveShapeUpToParent() {
+        WiresContainer parent = getParentPickerControl().getParent();
+        WiresShapeControlUtils.moveShapeUpToParent(getShape(), parent);
     }
+
+
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresShapeControlImpl.java]: useless empty lines

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/AbstractWiresControl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        return doMove(dx, dy);
+    }
+
+    protected boolean doMove(double dx, double dy) {
+        return false;
+    }
+
+    @Override
+    public boolean onMoveComplete() {
```
[the remark this answers is not in the document]
Comment (romartin) [reply src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/AbstractWiresControl.java]: yeahh agree :+1:  So taking the straightforward direction and just refactoring to a `void` return type on the signature for the [onMoveComplete](https://github.com/kiegroup/lienzo-core/blob/master/src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/WiresMoveControl.java#L32), this way code will look simpler as well. thx!

**Comment by romartin** [comment]
@hasys  @tiagodolphine  Fixed issues from your review's feedback. Thanks!

**Comment by romartin** [comment]
@hasys @tiagodolphine fixed an issue reported on the review and rebased

**Pull request merged**